public/components/Callback.js

```diff
@@ -1,21 +1,63 @@
 import React from 'react'
 import { Container } from 'semantic-ui-react'
-
 import '../styles/reset.css'
 import '../styles/main.css'
+import { getData } from "../utils/getData";
 
 class Callback extends React.Component {
+
   errorMessage = "Please be patient, you will be logged in."
+  
+  getPermissions = (token) => {
+    this.setState({loading: true, error: null});
+    if (process.env.PERMISSIONS_DISABLED === 'true'){
+      this.checkSuccess()
+    } else {
+      getData(process.env.API_URL + "/api/v1.0/auth/permissions", token)
+      .then(data => {
+        localStorage.setItem('permissions', JSON.stringify(data))
+        this.errorMessage = "Permissions are retrieved."
+        this.checkSuccess()
+      })
+      .catch((error) => {
+        this.errorMessage = "There is an error with collecting the permissions. Please try to reload this page or login again."
+        console.log(error)
+      }); 
+    }
+  };
 
-  componentDidMount () {
-    // First check if logged in
-    if (localStorage.getItem('token') !== null) {
+  parseJwt = (token)  =>{
+    const base64Url = token.split('.')[1];
+    const base64 = base64Url.replace(/-/g, '+').replace(/_/g, '/');
+    const jsonPayload = decodeURIComponent(window.atob(base64).split('').map(function(c) {
+        return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2);
+    }).join(''));
+    return JSON.parse(jsonPayload);
```

Comment:
I think there is a helper for this in jwt-decode that we're using
```
import { jwtDecode } from "jwt-decode";
const token = "eyJ0eXAiO.../// jwt token";
const decoded = jwtDecode(token);
```
source: https://www.npmjs.com/package/jwt-decode

worth checking out at least :)